File: domain/src/main/java/com/arturogutierrez/openticator/domain/bootstrap/StorageInitializator.java
package com.arturogutierrez.openticator.domain.bootstrap;

import com.arturogutierrez.openticator.domain.DatabaseConfigurator;
import com.arturogutierrez.openticator.domain.Preferences;
import javax.inject.Inject;

public class StorageInitializator {

  private final Preferences preferences;
  private final DatabaseConfigurator databaseConfigurator;

  @Inject
  public StorageInitializator(Preferences preferences, DatabaseConfigurator databaseConfigurator) {
    this.preferences = preferences;
    this.databaseConfigurator = databaseConfigurator;
  }

  public void configureStorage() {
    String passwordInBase64 = preferences.getMasterPassword();
    if (passwordInBase64 != null) {
      databaseConfigurator.configure(passwordInBase64);
    }
  }
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/bootstrap/StorageInitializator.kt
package com.arturogutierrez.openticator.domain.bootstrap

import com.arturogutierrez.openticator.storage.database.DatabaseConfigurator
import com.arturogutierrez.openticator.storage.preferences.Preferences
import javax.inject.Inject

class StorageInitializator @Inject constructor(val preferences: Preferences, val databaseConfigurator: DatabaseConfigurator) {

  fun configureStorage() {
    preferences.masterPassword?.let {
      databaseConfigurator.configure(it)
    }
  }
}
